### PR TITLE
vpd-tool: Fix fixSystemVPD option issues.

### PR DESCRIPTION
**Pull request by GiridhariKrishnan**
vpd-tool fixSystemVPD option update following fixes are made:
1) fix for input error, for which switch case is giving default value and doesn't exit.
2) Display format is corrected with this change,
3) Hexadecimal string display change due to https://github.com/ibm-openbmc/openpower-vpd-parser/pull/143 change getting in before this.

Initial Error scenario:
In Following case "Data On BMC" is in ASCII for some, and Retry for exit input.

S.No  Record  Keyword  Data On BMC                                                                Data On System Backplane                                                   Data Mismatch
     VSYS    WN       C050760BD482                                                               0x433035303736304244343832                     YES                                   
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

     VSYS    RG       0x00000000                                                                 0x00000000                                                                 NO            
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Enter 1 => If you choose the data on BMC for all mismatching record-keyword pairs
Enter 2 => If you choose the data on System Backplane for all mismatching record-keyword pairs
Enter 3 => If you wish to explore more options
Enter 0 => To exit successfully : 0

===============================================================================================================================================================================================

Provide a valid option. Retry.

Test:
Check the function as a whole.
vpd-tool --fixSystemVPD
Enter 0 => To exit successfully : 0

===============================================================================================================================================================================================

Exit successfully
2) second operation check
:~# busctl set-property xyz.openbmc_project.Inventory.Manager /xyz/openbmc_project/inventory/system/chassis/motherboard com.ibm.ipzvpd.VSYS "BR" ay 2 75 40
5     VSYS    BR       0x4b28                                                                     0x4a30                                                                     YES
Enter 1 => If you choose the data on BMC for all mismatching record-keyword pairs
Enter 2 => If you choose the data on System Backplane for all mismatching record-keyword pairs
Enter 3 => If you wish to explore more options
Enter 0 => To exit successfully : 2
O/P: ~# vpd-tool --fixSystemVPD

Restorable record-keyword pairs and their data on BMC & System Backplane.

===============================================================================================================================================================================================
S.No  Record  Keyword  Data On BMC                                                                Data On System Backplane                                                   Data Mismatch
===================================================================================-------------------------------------------------------------------------------
5     VSYS    BR       0x4a30                                                                     0x4a30                                                                     NO     
===================================================================================

3) More options [3] -> option 6 and 7 check.
Enter 3 => If you wish to explore more options
Enter 0 => To exit successfully : 3

===============================================================================================================================================================================================

Iterate through all restorable record-keyword pairs
===============================================================================================================================================================================================
S.No  Record  Keyword  Backup Data                                                                Primary Data                                                               Data Mismatch 
1     UTIL    D0       0x02                                                                       0x02                                                                      NO            
===============================================================================================================================================================================================

No mismatch found.

Enter 6 => If you wish to enter a new value to update both on BMC and System Backplane
Enter 7 => If you wish to skip the above record-keyword pair
Enter 0 => To exit successfully : 9

===============================================================================================================================================================================================

Provide a valid option. Retrying for the current record-keyword pair

===============================================================================================================================================================================================
S.No  Record  Keyword  Backup Data                                                                Primary Data                                                               Data Mismatch 
1     UTIL    D0       0x02                                                                       0x02                                                                      NO            
=================================================================================

===============================================================================================================================================================================================
S.No  Record  Keyword  Data On BMC                                                                Data On System Backplane                                                   Data Mismatch 
1     UTIL    D0       0x02                                                                       0x02                                                                       NO            
===============================================================================================================================================================================================

No mismatch found.

Enter 6 => If you wish to enter a new value to update both on BMC and System Backplane
Enter 7 => If you wish to skip the above record-keyword pair
Enter 0 => To exit successfully : 6

===============================================================================================================================================================================================

Enter the new value to update both on BMC & System Backplane (Value should be in ASCII or in HEX(prefixed with 0x)) : 0x01

Data updated successfully.

===============================================================================================================================================================================================
S.No  Record  Keyword  Data On BMC                                                                Data On System Backplane                                                   Data Mismatch 
2     LXR0    LX       0x3100080100300074                                                         0x3100080100300074                                                         NO            
===============================================================================================================================================================================================

No mismatch found.

Enter 6 => If you wish to enter a new value to update both on BMC and System Backplane
Enter 7 => If you wish to skip the above record-keyword pair
Enter 0 => To exit successfully : 7

===============================================================================================================================================================================================

Skipped the above record-keyword pair. Continue to the next available pair.

~# vpd-tool --fixSystemVPD

Restorable record-keyword pairs and their data on BMC & System Backplane.

S.No  Record  Keyword  Data On BMC                                                                Data On System Backplane                                                   Data Mismatch
===============================================================================================================================================================================================
1     UTIL    D0       0x01                                                                       0x01                                                                       NO            
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Enter 6 => If you wish to enter a new value to update both on BMC and System Backplane
Enter 7 => If you wish to skip the above record-keyword pair
Enter 0 => To exit successfully : 0

===============================================================================================================================================================================================

Exit successfully

In all cases exiting successfully.

### CODE REVIEW

File: vpd_tool_impl.cpp [hunk 1]
```diff
@@ -914,6 +913,90 @@ void VpdTool::parseSVPDOptions(const nlohmann::json& json)
                            "record-keyword pair. Exit successfully.\n";
                 }
 
+                exit(0);
```
Comment (jinuthomas) [inline vpd_tool_impl.cpp:916]: why is this case moved above ? can you revert this so that the diff is better

File: vpd_tool_impl.cpp [hunk 2]
```diff
@@ -942,99 +1025,6 @@ void VpdTool::parseSVPDOptions(const nlohmann::json& json)
...
-                                cout
-                                    << "\nSkipped the above record-keyword "
-                                       "pair. "
-                                       "Continue to the next available pair.\n";
-                                break;
-                            case VpdTool::EXIT:
-                                cout << "\nExit successfully\n";
-                                exit(0);
-                            default:
-                                cout << "\nProvide a valid option. Retrying "
```
Comment (jinuthomas) [inline vpd_tool_impl.cpp:1032]: don't we need a break in default case

File: vpd_tool_impl.cpp (hunk 1 above)
Comment (Alpana07) [reply vpd_tool_impl.cpp:916]: Right, it is showing too much of difference. and not getting what is the actual change here.
Please revert this.

**Review by Alpana07 — CHANGES_REQUESTED** [review]

**Review by jinuthomas — CHANGES_REQUESTED** [review]
Any update on the review comments ?

File: vpd_tool_impl.cpp (hunk 2 above)
Comment (GiridhariKrishna) [reply vpd_tool_impl.cpp:1032]: I discussed with Priyanga, it seems to be not intended to have break here.

File: vpd_tool_impl.cpp (hunk 1 above)
Comment (GiridhariKrishna) [reply vpd_tool_impl.cpp:916]: I'll revert it.

File: vpd_tool_impl.cpp [hunk 3]
```diff
@@ -949,91 +948,82 @@ void VpdTool::parseSVPDOptions(const nlohmann::json& json)
 
                 for (const auto& data : recKwData)
                 {
-                    do
```
Comment (PriyangaRamasamy) [inline vpd_tool_impl.cpp:952]: do not remove do while loop as it hits the default case. So change the switch case at line 997 to if else.

File: vpd_tool_impl.cpp [hunk 4]
```diff
@@ -955,12 +954,13 @@ void VpdTool::parseSVPDOptions(const nlohmann::json& json)
 
                         cout << left << setw(6) << "S.No" << left << setw(8)
                              << "Record" << left << setw(9) << "Keyword" << left
-                             << setw(75) << setfill(' ') << "Data On BMC"
+                             << setw(75) << setfill(' ') << "Backup Data"
```
Comment (PriyangaRamasamy) [inline vpd_tool_impl.cpp:957]: make this change wherever required in this file

File: vpd_tool_impl.cpp [hunk 5]
```diff
@@ -989,49 +989,52 @@ void VpdTool::parseSVPDOptions(const nlohmann::json& json)
 
                         cin >> option;
                         cout << '\n' << outline << endl;
-
                         EditorImpl edit(constants::systemVpdFilePath, json,
                                         get<1>(data), get<2>(data));
-                        string value;
-
-                        switch (option)
+                        if (option == VpdTool::BMC_DATA_FOR_CURRENT)
```
Comment (PriyangaRamasamy) [inline vpd_tool_impl.cpp:994]: can you change this to BACKUP_DATA_FOR_CURRENT
and instead of BMC_DATA_FOR_ALL change it to BACKUP_DATA_FOR_ALL

**Review by PriyangaRamasamy — COMMENTED** [review]
ci is failing for clang format. please fix it

File: vpd_tool_impl.cpp (hunk 5 above)
Comment (GiridhariKrishnan) [reply vpd_tool_impl.cpp:994]: It won't have effect on the display, so may not be required, I think. As it's downstream code and variables have been approved before.

Comment (PriyangaRamasamy) [reply vpd_tool_impl.cpp:994]: this change is required to have the naming convention consistent all over the code

Comment (GiridhariKrishna) [reply vpd_tool_impl.cpp:993]: Okay, I'll change but not sure though. Thanks.

**Review by PriyangaRamasamy — APPROVED** [review]

**Pull request merged**